### PR TITLE
Feature/task1

### PR DESCRIPTION
**Pull request by m4qo5**

### CODE REVIEW

File: repeating_item.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        if not dict1:
+            return 0
+    except:
+        raise TypeError
+    if len(set(dict1.values())) == 1:
+        return 0
+    else:
+        max_value = max(dict1, key=dict1.get)
+        return max_value
+
```
Comment (MaksimLion) [inline repeating_item.py]: extra block else

File: length_str.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+def length_str(str_list):
+    d = dict()
```
Comment (AlehSivakou) [inline length_str.py]: please avoid short naming. naming 'd' is not an option. the same as for k,v

File: length_str.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    mas_output = []
+    for item in str_list:
```
Comment (AlehSivakou) [inline length_str.py]: please provide elegant and Python-specific way to write this 'for-loop' block of code using.

File: length_str.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+        d[len(item)] = item
+    for k,v in sorted(d.items()):
+        mas_output.append(v)
```
Comment (AlehSivakou) [inline length_str.py]: please provide elegant and Python-specific way to write this 'for-loop' block of code using.

File: repeating_item.py (hunk 1 above)
Comment (AlehSivakou) [reply repeating_item.py]: there is no need to define variable unless you have more that one rows before returning value. In other words you can write this as 'return max(dict1, key=dict1.get)'

File: sort.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+def bubble_sort(list):
+    for i in range(len(list)):
```
Comment (AlehSivakou) [inline sort.py]: Please use more elegant Python-specific way to write nested for loops.

File: year.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        return True
+    elif num % 100 == 0 and num % 400 !=0:
+        return False
+    elif num % 4 == 0:
+        return True
+    else:
+        return False
+        
+
+from calendar import isleap
```
Comment (AlehSivakou) [inline year.py]: Please follow PEP-8.

File: year.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+def year(num):
+    if num % 400 == 0:
```
Comment (AlehSivakou) [inline year.py]: Please minimize the number of if-else conditions. As an option try to use logical operators like 'or' and 'and'.

File: length_str.py (hunk 2 above)
Comment (m4qo5) [reply length_str.py]: done

File: length_str.py (hunk 3 above)
Comment (m4qo5) [reply length_str.py]: done

File: length_str.py (hunk 4 above)
Comment (m4qo5) [reply length_str.py]: done

File: year.py (hunk 6 above)
Comment (m4qo5) [reply year.py]: done

File: year.py (hunk 7 above)
Comment (m4qo5) [reply year.py]: done

**Pull request merged**